controlplane/kubeadm/controllers/scale.go

```diff
@@ -64,8 +64,8 @@ func (r *KubeadmControlPlaneReconciler) scaleUpControlPlane(ctx context.Context,
 	logger := controlPlane.Logger()
 
 	// reconcileHealth returns err if there is a machine being delete which is a required condition to check before scaling up
-	if err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); err != nil {
-		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
+	if result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); !result.IsZero() || err != nil {
+		return result, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
```

Comment:
Do you still want this returning a RequeueAfterError?

Reply:
No, good catch. This means that this error was shadowing everything else too.